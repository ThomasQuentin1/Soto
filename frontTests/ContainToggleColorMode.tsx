import React from 'react';
import ToggleColorMode from '../components/settings/ToggleColorMode';
import { useDarkMode } from "../components/settings/useDarkMode";

const ContainToggleColorMode = () => {
  const [theme, setTheme] = useDarkMode();

    return (<ToggleColorMode theme={theme} toggleTheme={setTheme}></ToggleColorMode>)
}

export default ContainToggleColorMode;