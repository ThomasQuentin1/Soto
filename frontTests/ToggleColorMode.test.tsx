import React from 'react';
import renderer from 'react-test-renderer';
import ContainToggleColorMode from './ContainToggleColorMode';

test('DarkModeParent is a container that apply the theming and style to all childs, it has to be used in pages', () => {
    const component = renderer.create(
        <ContainToggleColorMode></ContainToggleColorMode>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});