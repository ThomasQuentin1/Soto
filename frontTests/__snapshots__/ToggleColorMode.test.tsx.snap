// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeParent is a container that apply the theming and style to all childs, it has to be used in pages 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <img
    alt="lightMode"
    src="/images/light/sun.png"
    style={
      Object {
        "height": "25px",
        "width": "25px",
      }
    }
  />
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          aria-label="secondary checkbox"
          checked={false}
          className="PrivateSwitchBase-input-4 MuiSwitch-input"
          name="colorMode"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
  <img
    alt="darkMode"
    src="/images/light/moon.png"
    style={
      Object {
        "height": "25px",
        "width": "25px",
      }
    }
  />
</div>
`;
